src/round.rs

```diff
@@ -85,14 +92,155 @@ fn span_for_digits(digits: u16) -> u32 {
     }
 }
 
+/// Extension trait for rounding or truncating a DateTime by a Duration.
+///
+/// # Limitations
+/// Both rounding and truncating are done via [`Duration::num_nanoseconds`] and
+/// [`DateTime::timestamp_nanos`]. This means that they will fail if either the
+/// `Duration` or the `DateTime` are too big to represented as nanoseconds. They
+/// will also fail if the `Duration` is bigger than the timestamp.
+pub trait DurationRound: Sized {
+    /// Error that can occur in rounding or truncating
+    #[cfg(any(feature = "std", test))]
+    type Err: std::error::Error;
+
+    /// Error that can occur in rounding or truncating
+    #[cfg(not(any(feature = "std", test)))]
+    type Err;
```

Comment:
I think it's probably worth adding a trait bound here of `Debug + Display`, [anyhow uses that bound in no std](https://github.com/dtolnay/anyhow/blob/1089f316136a9718b6e9b5943772822958cd268d/src/error.rs#L70) so it seems not insane. We can compatibly remove trait bounds down the line, but not add them, so I'd rather be as conservative as possible here, but I'm not familiar with no_std in general.